### PR TITLE
CSCAP-38 Add some unit testing framework

### PR DESCRIPTION
**Pull request by Xerasin**
Accidently worked on CSCAP-38, wanted to get this started before my main work so I could write tests as I go. 

I put tests in backend/src/test/java/com/sim_backend as it's I believe it's preferred for jacoco. 
This just adds JUnit, Mockito, Jacoco and a Jacoco workflow to the repository.

### CODE REVIEW

File: backend/src/main/java/com/sim_backend/Main.java [hunk 1]
```diff
@@ -33,6 +33,6 @@ public static void main(final String[] args) {
         }).start(backendPort); // Start the server
 
         // Register a test route
-        TestMessageController.registerRoutes(app);
+        //TestMessageController.registerRoutes(app);
```
Comment (pglob) [inline backend/src/main/java/com/sim_backend/Main.java:36]: Was commenting this out intentional?

File: backend/Dockerfile [hunk 2]
```diff
@@ -14,7 +14,7 @@ WORKDIR /app
 RUN mvn clean package -DskipTests
 
 # Stage 2: Run the application
-FROM eclipse-temurin:21-jdk-jammy
+FROM build as run
```
Comment (pglob) [inline backend/Dockerfile:17]: Minor: I made this mistake too, but the different casing on "FROM" and "as" causes warnings. Please change "as" to "AS". Instances on lines 4, 17, and 29.

**Comment by pglob** [comment]
- [ ] Make sure commits are squashed before merging.

File: backend/src/main/java/com/sim_backend/Main.java (hunk 1 above)
Comment (Xerasin) [reply backend/src/main/java/com/sim_backend/Main.java]: I meant to revert this, I just needed a change in Main.java to make sure the workflow is actually picking up on the changed files.

**Review by pglob — APPROVED** [review]

**Pull request merged**